### ISSUE TITLE
Lack of documentation

### ISSUE THREAD

**Issue by PixelOcelot** [decibel/variant#3]
Hello.
I'm sorry, maybe it's not a functional issue, but it causes some trouble with using your type.
Could you please make step-by-step quickstart guide? 
With emphasis on what is default variant, how to register non-default variant and allow storage for that.
Maybe simple use cases (insert, update, using different value types for same record, etc.).
That would be great.

**Comment by PixelOcelot** [comment]
I've figured out how to register variant and now I see that documentation covers it(partially, regtypes and storage allowed not covered), but to be more clear it could be rephrased and converted to step-by-step. Now it's default variant used for examples, though latter it was changed, I think it would be better to register new variant and show examples with it.

**Comment by decibel** [comment]
Yeah, I need to redo some of the docs. The latest version fundamentally changes how some of these things work. In particular, if a variant doesn't allow storage and you don't specify any allowed types, it will allow ALL types. That's how the DEFAULT variant is setup now, and it's also enabled by default.

BTW, the reason for all these games is to protect from someone doing

CREATE TABLE v(v variant);
INSERT INTO v VALUES( ...::my_custom_type );
DROP TYPE my_customer_type;

which would result in no longer being able to retrieve the value stored in the table, even if you re-installed my_custom_type.
